online_party_system/party_match.go

```diff
@@ -243,6 +310,28 @@ func (p *PartyMatch) MatchJoinAttempt(ctx context.Context, logger runtime.Logger
 	// Ensure join metadata is stored to be used in MatchJoin callback.
 	s.joinMetadata[presence.GetUserId()] = p.config.MatchJoinMetadataFilter(metadata)
 
+	rebuildBlockedUsersIfNeeded(ctx, logger, nk, s)
+
+	// Test if anyone "in" the party has blocked this user
+	if _, ok := s.blockedUsers[presence.GetUserId()]; ok {
+		return s, false, "Someone in this party blocked you (re-phrase this)"
+	}
+
+	// Test if this user has blocked anyone invited to this party
+	blockedSetFromUser := getUsersBlockedFriends(ctx, logger, nk, presence.GetUserId())
```

Comment:
Since we've loaded the user's blocked friend relations anyway, maybe best to keep them instead of going back to database after (and only if) their join is approved?

Reply:
yup